RFS/src/main/java/com/rfs/ReindexFromSnapshot.java

```diff
@@ -0,0 +1,298 @@
+            repo = new S3Repo(s3LocalDirPath, s3RepoUri, s3Region);
+        } else {
+            throw new IllegalArgumentException("You must specify either a snapshot directory or an S3 URI");
+        }
+
+        // Set the transformer
+        Transformer transformer = TransformFunctions.getTransformer(sourceVersion, targetVersion, awarenessAttributeDimensionality);
+
+        try {
+            // ==========================================================================================================
+            // Read the Repo data file
+            // ==========================================================================================================
+            logger.info("==================================================================");
+            logger.info("Attempting to read Repo data file...");
+            SnapshotRepo.Provider repoDataProvider;
+            if (sourceVersion == ClusterVersion.ES_6_8) {
+                repoDataProvider = new SnapshotRepoProvider_ES_6_8(repo);
+            } else {
+                repoDataProvider = new SnapshotRepoProvider_ES_7_10(repo);
+            }
+
+            if (repoDataProvider.getSnapshots().size() > 1){
+                // Avoid having to deal with things like incremental snapshots
+                throw new IllegalArgumentException("Only repos with a single snapshot are supported at this time");
+            }
+
+            logger.info("Repo data read successfully");
+
+            // ==========================================================================================================
+            // Read the Snapshot details
+            // ==========================================================================================================
+            logger.info("==================================================================");
+            logger.info("Attempting to read Snapshot details...");
+            String snapshotIdString = repoDataProvider.getSnapshotId(snapshotName);
+
+            if (snapshotIdString == null) {
+                logger.warn("Snapshot not found");
+                return;
+            }
+            SnapshotMetadata.Data snapshotMetadata;
+            if (sourceVersion == ClusterVersion.ES_6_8) {
+                snapshotMetadata = new SnapshotMetadataFactory_ES_6_8().fromRepo(repo, repoDataProvider, snapshotName);
+            } else {
+                snapshotMetadata = new SnapshotMetadataFactory_ES_7_10().fromRepo(repo, repoDataProvider, snapshotName);
+            }
+            logger.info("Snapshot data read successfully");
+
+            if (!snapshotMetadata.isIncludeGlobalState() && ((movementType == MovementType.EVERYTHING) || (movementType == MovementType.METADATA))){
+                throw new IllegalArgumentException("Snapshot does not include global state, so we can't move metadata");
+            }
+
+            if (!"SUCCESS".equals(snapshotMetadata.getState())){
```

Comment:
Static would be nice here or ENUM if we plan to check other states in future

Reply:
Done